hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMFileCreateRequestV1.java

```diff
@@ -0,0 +1,289 @@
+      }
+
+      // add all missing parents to dir table
+      missingParentInfos =
+              OMDirectoryCreateRequestV1.getAllMissingParentDirInfo(
+                      ozoneManager, keyArgs, pathInfoV1, trxnLogIndex);
+
+      // total number of keys created.
+      numKeysCreated = missingParentInfos.size();
+
+      // do open key
+      OmBucketInfo bucketInfo = omMetadataManager.getBucketTable().get(
+          omMetadataManager.getBucketKey(volumeName, bucketName));
+
+      OmKeyInfo omFileInfo = prepareFileInfo(omMetadataManager, keyArgs,
+              dbFileInfo, keyArgs.getDataSize(), locations,
+              getFileEncryptionInfo(keyArgs), ozoneManager.getPrefixManager(),
+              bucketInfo, pathInfoV1, trxnLogIndex,
+              ozoneManager.isRatisEnabled());
+
+      long openVersion = omFileInfo.getLatestVersionLocations().getVersion();
+      long clientID = createFileRequest.getClientID();
+      String dbOpenFileName = omMetadataManager.getOpenFileName(
+              pathInfoV1.getLastKnownParentId(), pathInfoV1.getLeafNodeName(),
+              clientID);
+
+      // Append new blocks
+      List<OmKeyLocationInfo> newLocationList = keyArgs.getKeyLocationsList()
+          .stream().map(OmKeyLocationInfo::getFromProtobuf)
+          .collect(Collectors.toList());
+      omFileInfo.appendNewBlocks(newLocationList, false);
+
+      omVolumeArgs = getVolumeInfo(omMetadataManager, volumeName);
+      omBucketInfo = getBucketInfo(omMetadataManager, volumeName, bucketName);
+      // check volume quota
+      long preAllocatedSpace = newLocationList.size()
+          * ozoneManager.getScmBlockSize()
+          * omFileInfo.getFactor().getNumber();
+      checkVolumeQuotaInBytes(omVolumeArgs, preAllocatedSpace);
+
+      // Add to cache entry can be done outside of lock for this openKey.
+      // Even if bucket gets deleted, when commitKey we shall identify if
+      // bucket gets deleted.
+      omMetadataManager.getOpenKeyTable().addCacheEntry(
+          new CacheKey<>(dbOpenFileName),
+          new CacheValue<>(Optional.of(omFileInfo), trxnLogIndex));
+
+      // Add cache entries for the prefix directories.
+      // Skip adding for the file key itself, until Key Commit.
+      OMFileRequest.addDirectoryTableCacheEntries(omMetadataManager,
+              Optional.absent(), Optional.of(missingParentInfos),
+              trxnLogIndex);
+
+      // update usedBytes atomically.
+      omVolumeArgs.getUsedBytes().add(preAllocatedSpace);
+      omBucketInfo.getUsedBytes().add(preAllocatedSpace);
+
+      // Prepare response
+      omResponse.setCreateFileResponse(CreateFileResponse.newBuilder()
+          .setKeyInfo(omFileInfo.getProtobuf())
+          .setID(clientID)
+          .setOpenVersion(openVersion).build())
+          .setCmdType(Type.CreateFile);
+      omClientResponse = new OMFileCreateResponseV1(omResponse.build(),
+          omFileInfo, missingParentInfos, clientID, omVolumeArgs, omBucketInfo);
```

Comment:
OMKeyInfo is exposed to Ozone client, so I have kept 'keyName' as it is. Anyway fileName can be extracted from the  keyName so that I thought of reducing the load on DB.

Hope this make sense to you.

Reply:
But now OmKeyInfo has entire keyPath which is not really needed to be persisted to DB in this new V1 Request.

Reply:
I completely agree with you that, V1 need only the fileName in DB.

On a second thought, do we need to maintain two different `OMKeyInfo#keyName` semantics in DB. For 100% S3 , 'keyName' stores full path. For 100% FS, 'keyName' stores only the fileName in the value part. 
```
message keyInfo {
       ....
       required string keyName = 3;
```

Reply:
Yes. Because if we store the entire keyPath in DB when rename, then if we change the only parentObjectID and then leave keyName then they will diverge.

And also this is FS API not called from S3.

Reply:
Fully agreed with you, thanks for pointing out the rename case. 

How about store `fileName` in the `keyName` field in DB. Then, while populating OmKeyInfo logically will prepare fullPath in `keyName` and will store `fileName` separately.

Reply:
Yes, you read my mind :)
It works perfectly.

Reply:
Done the changes in new commit

Reply:
Added to the cache correctly(With KeyName as leaf node name), but in response class, we passed omKeyInfo this should be also the same thing which we have added to the cache right?

Reply:
Inside OMFileCreateResponseV1#addToDBBatch() method has the actual call to the DB, it is doing the changes in utility OMFileRequest function. I have done this way to ensure that, only when updating the TableCache and persisting to the Table it makes the DB representation as `keyName=leafNodeName`. All the other places(any functional checks inside OM, logging inside OM, client response etc..) will have the `keyName=fullPath` and `fileName=leafNodeName`.

   ```
 OMFileRequest.addToOpenFileTable(omMetadataMgr, batchOp, getOmKeyInfo(),
            getOpenKeySessionID());
```

Also, for better code maintenance, I will be adding these conversion in `OMFileRequest` utility functions. Hope this is fine for you?

Reply:
With the current approach in PR looks like we do unnecessarily one more time copy object. What we add to the cache, if we pass the same instance to Response class then same can be persisted. With this way cache and Response classes uses the same instance, and anyway in logging in createRequest we use Keyname, I see no issue over there.

Reply:
Sure, updated patch addressing this point. Thanks!